watchdog/src/health.rs

```diff
@@ -117,14 +144,69 @@ mod test {
 
     #[test]
     fn test_median() {
-        assert_eq!(median(vec![]), None);
-        assert_eq!(median(vec![1]), Some(1));
-        assert_eq!(median(vec![2, 1]), Some(1));
-        assert_eq!(median(vec![3, 2, 1]), Some(2));
-        assert_eq!(median(vec![4, 3, 2, 1]), Some(2));
-        assert_eq!(median(vec![5, 4, 3, 2, 1]), Some(3));
-        assert_eq!(median(vec![20, 20, 10, 10]), Some(15));
-        assert_eq!(median(vec![20, 15, 10]), Some(15));
+        assert_eq!(median(&[]), None);
+        assert_eq!(median(&[1]), Some(1));
+        assert_eq!(median(&[2, 1]), Some(1));
+        assert_eq!(median(&[3, 2, 1]), Some(2));
+        assert_eq!(median(&[4, 3, 2, 1]), Some(2));
+        assert_eq!(median(&[5, 4, 3, 2, 1]), Some(3));
+        assert_eq!(median(&[20, 20, 10, 10]), Some(15));
+        assert_eq!(median(&[20, 15, 10]), Some(15));
+    }
+
+    /// Test data for `calculate_height_target`.
+    struct CalculateHeightTargetTestData {
+        heights: &'static [u64],
+        min_explorers: usize,
+        blocks_behind_threshold: i64,
+        blocks_ahead_threshold: i64,
+        expected: Option<u64>,
+    }
+
+    /// Tests `calculate_height_target` with the given test data.
+    fn test_calculate_height_target(params: CalculateHeightTargetTestData) {
+        assert_eq!(
+            calculate_height_target(
+                params.heights,
+                params.min_explorers,
+                params.blocks_behind_threshold,
+                params.blocks_ahead_threshold
+            ),
+            params.expected
+        );
+    }
+
+    #[test]
+    fn test_calculate_height_target_not_enough_explorers() {
+        test_calculate_height_target(CalculateHeightTargetTestData {
+            heights: &[10],
+            min_explorers: 3,
+            blocks_behind_threshold: -1,
+            blocks_ahead_threshold: 1,
+            expected: None,
+        });
+    }
```

Comment:
NIT: Are `CalculateHeightTargetTestData` and `test_calculate_height_target()`necessary at all? I do not see that we make code shorter using them.

Reply:
It makes tests a bit more readable when you see the names of variables.

```rs
// Before.
#[test]
fn test_calculate_height_target_explorers_not_in_range() {
    assert_eq!(test_calculate_height_target(&[10, 20, 30], 3, -1, 1), None);
}

// After.
#[test]
fn test_calculate_height_target_explorers_not_in_range() {
    test_calculate_height_target(CalculateHeightTargetTestData {
        heights: &[10, 20, 30],
        min_explorers: 3,
        blocks_behind_threshold: -1,
        blocks_ahead_threshold: 1,
        expected: None,
    });
}
```